src/components/Hoverable/index.js

```diff
@@ -1,189 +1,211 @@
-        }
+        document.addEventListener('mouseover', unsetHoveredIfOutside);
 
-        this.setIsHovered(false);
-    }
+        return () => document.removeEventListener('mouseover', unsetHoveredIfOutside);
+    }, [unsetHoveredIfOutside]);
 
-    handleVisibilityChange() {
-        if (document.visibilityState !== 'hidden') {
+    useEffect(() => {
+        if (!disabled || !isHovered) {
             return;
         }
+        setIsHovered(false);
+    }, [disabled, isHovered]);
 
-        this.setIsHovered(false);
-    }
-
-    render() {
-        let child = this.props.children;
-        if (_.isArray(this.props.children) && this.props.children.length === 1) {
-            child = this.props.children[0];
+    useEffect(() => {
+        if (disabled) {
+            return;
         }
-
-        if (_.isFunction(child)) {
-            child = child(this.state.isHovered);
+        if (onHoverIn && isHovered) {
+            return onHoverIn();
         }
-
-        if (!DeviceCapabilities.hasHoverSupport()) {
-            return child;
+        if (onHoverOut && !isHovered) {
+            return onHoverOut();
         }
-
-        return React.cloneElement(React.Children.only(child), {
-            ref: (el) => {
-                this.wrapperView = el;
-
-                // Call the original ref, if any
-                const {ref} = child;
-                if (_.isFunction(ref)) {
-                    ref(el);
-                    return;
-                }
-
-                if (_.isObject(ref)) {
-                    ref.current = el;
-                }
-            },
-            onMouseEnter: (el) => {
-                this.setIsHovered(true);
-
-                if (_.isFunction(child.props.onMouseEnter)) {
-                    child.props.onMouseEnter(el);
-                }
-            },
-            onMouseLeave: (el) => {
-                this.setIsHovered(false);
-
-                if (_.isFunction(child.props.onMouseLeave)) {
-                    child.props.onMouseLeave(el);
-                }
-            },
-            onBlur: (el) => {
-                // Check if the blur event occurred due to clicking outside the element
-                // and the wrapperView contains the element that caused the blur and reset isHovered
-                if (!this.wrapperView.contains(el.target) && !this.wrapperView.contains(el.relatedTarget)) {
-                    this.setIsHovered(false);
-                }
-
-                if (_.isFunction(child.props.onBlur)) {
-                    child.props.onBlur(el);
-                }
-            },
-        });
+    }, [disabled, isHovered, onHoverIn, onHoverOut]);
+
+    // Expose inner ref to parent through outerRef. This enable us to use ref both in parent and child.
+    useImperativeHandle(outerRef, () => ref.current, []);
+
+    const child = useMemo(() => React.Children.only(mapChildren(children, isHovered)), [children, isHovered]);
+
+    const onMouseEnter = useCallback(
```

Comment:
The method should be named for what it does and never what callback they handle, the same for onMouseLeave and onBlur

> any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. toggleReport and not onIconClick).

Reply:
Done ✅